fixed zammad no jquery (Issue 2561)

blagisquet opened the pull request:
Relative to: https://github.com/zammad/zammad/issues/2561

Hello, 

I wanted to use Zammad without jQuery and had to fix few things to make it works (see issue 2561).
I fixed button relative problem and just added a space between + and px which cause upload files not working.

Hope you'll like it.

Baptiste Lagisquet

Commit:
fixed zammad no jquery

Commit:
fixed renderBase

Commit:
spelling mistake

thorsteneckel commented:
Hey @blagisquet - thanks a lot for your contribution! Could you please elaborate what the issue was and how you solved it and why you chose the way you did it? Looking forward to merge it.

blagisquet commented:
Hi, the problem was that when you used zammad without jQuery if there was no button as there was no if statement zammad returns undefined. So I just added if  statement (lines 470-472 and lines 777-779 on public/assets/chat/chat-no-jquery.coffee). 

There was also a problem with concatenation (img.style.maxWidth = width + 'px') I just added a space (lines 554 and 602 on public/assets/chat/chat-no-jquery.coffee ). And I deleted the +’px’ from (@scrollRoot.scrollTop = @rootScrollOffset +'px') as it wasn’t needed.

Finally, I updated the method for render base cause it was relaunching all the DOM when zammad started after the render of Angular (lines 489-494  on public/assets/chat/chat-no-jquery.coffee ).

I hope everything is clear enough, it’s my first PR on an external project. :)

mantas commented:
Hi @blagisquet ,

Thank you for your contribution!

It looks like this issue was addressed while we got to your PR. Feel free to test `develop` branch and reopen this PR if the issue persists.

Apologies for the delay.
